src/engraving/io/mscreader.cpp

```diff
@@ -119,13 +119,16 @@ QByteArray MscReader::readStyleFile() const
 
 QByteArray MscReader::readScoreFile() const
 {
-    QString mscxFileName;
-    QStringList files = reader()->fileList();
-    for (const QString& name : files) {
-        // mscx file in the root dir
-        if (!name.contains("/") && name.endsWith(".mscx")) {
-            mscxFileName = name;
-            break;
+    QString mscxFileName = QFileInfo(m_params.filePath).completeBaseName() + ".mscx";
+    QByteArray data = fileData(mscxFileName);
+    if (data.isEmpty()) {

```

Comment:
when would this ever happen though? is the idea that you'd actually be able to select a directory name to open instead of an mscx file?

Reply:
I would like the ыcore file in the `mscz` archive to have a fixed name `score.mscx`. But right now, this cannot be changed, and I'm not 100% sure that this is a good idea.
So, this code is just in case

> is the idea that you'd actually be able to select a directory name to open instead of an mscx file?
I thought about it, but there is a technical limitation, the API of Qt allows us to make either a file selection or a directory selection, but not both at once. I would not like to do two different menu items for opening.
So, now it is done so that we select the `mscx` file, but in fact, we select the project directory (similar to how we open the C++ project in IDE)

Reply:
Ok, as long as the facility to open/import mscx files when there are multiple ones in a directory works as expected - I do that all the time with the unit test files now (and it's somewhat irritating that you can't easily update them by saving, which slightly unexpectedly creates a whole new directory,  but it's workable)